tests/assembly/issues/issue-126585.rs

```diff
@@ -0,0 +1,29 @@
+//@ assembly-output: emit-asm
+//@ only-x86_64
+//@ compile-flags: -Copt-level=s -Cllvm-args=-x86-asm-syntax=intel
+//@ min-llvm-version: 19
+
+// Test for #126585.
+// Ensure that we do not create subsequent labels and unnecessary `jmp` instructions.
+
+#![crate_type = "lib"]
+
+#[no_mangle]
+fn checked_div_round(a: u64, b: u64) -> Option<u64> {
```

Comment:
I think it would be better to check that the IR does not have the extra undef phi input.

Reply:
At least for this case, I think so.